File: onboarding.tool.client/src/pages/dashboard/branches/types/branch.ts
export interface branch {
  id: number | null;
  isDefault: boolean;
  name: string;
  salesEmail: string;
  salesPhone: string;
  lettingsEmail: string;
  lettingsPhone: string;
  address: address;
}

export interface address {
  id: number | null;
  address1: string;
  address2: string | null;
  address3: string | null;
  town: string;
  postCode1: string;
  postCode2: string;
}

File: Onboarding.Tool.Model/Data/Branch.cs
using System.Text.Json.Serialization;

namespace Onboarding.Tool.Model.Data;

public class Branch
{
    public int Id { get; set; }

    public required bool IsDefault { get; set; }

    public required string Name { get; set; }

    public required string SalesEmail { get; set; }

    public required string SalesPhone { get; set; }

    public required string LettingsEmail { get; set; }

    public required string LettingsPhone { get; set; }

    public ICollection<BranchSetting> Settings { get; set; } = new List<BranchSetting>();

    public ICollection<BranchBuildDistrict> Districts { get; set; } = new List<BranchBuildDistrict>();

    [JsonIgnore]
    public int? AddressId { get; set; }

    public Address? Address { get; set; }

    [JsonIgnore]
    public int InstanceId { get; set; }

    [JsonIgnore]
    public Instance? Instance { get; set; } = null!;
}
